publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java

```diff
@@ -353,8 +353,9 @@ public ImportCandidate updateImportCandidate(ImportCandidate importCandidate) th
         return updateResourceService.updateImportCandidate(importCandidate);
     }
 
-    public void unpublishPublication(Publication publication) throws BadRequestException {
-        if (!PUBLISHED.equals(publication.getStatus())) {
+    public void unpublishPublication(Publication publication) throws BadRequestException, NotFoundException {
+        var existingPublication = readResourceService.getPublication(publication);
+        if (!PUBLISHED.equals(existingPublication.getStatus())) {
             throw new BadRequestException(ONLY_PUBLISHED_PUBLICATIONS_CAN_BE_UNPUBLISHED_ERROR_MESSAGE);
         }
         var allTicketsForResource = fetchAllTicketsForResource(Resource.fromPublication(publication));
```

Comment:
unpublish method in resourceService has always used publication provided to method, I think it should fetch existing publication and check that status of existing publication is PUBLISHED